src/pydargs/__init__.py

```diff
@@ -47,7 +47,10 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, **kwargs: Any)
         An instance of tp.
     """
     namespace = _create_parser(tp, **kwargs).parse_args(args)
-    return _create_object(tp, namespace)
+    result = _create_object(tp, namespace)
+    if len(namespace.__dict__):
+        raise RuntimeError("Internal pydargs error: Some namespace arguments have not been consumed.")
```

Comment:
This might be a different topic.. I was talking about allowing "--something-unknown" to not crash the argparser
```
import argparse

parser = argparse.ArgumentParser(description='Process some integers.')
parser.add_argument('something', metavar='N', type=int, default=1)

args = parser.parse_args(["2", "--something-unknown"])
print(args)
```

So we could perhaps change it to `args, unknown = parser.parse_known_args()` 
I thought this would add those unknown arguments to the namespace, but this is not the case.  👍

Reply:
In our case this would always cause argparse to exit with an error message. As we do not expose the namespace to the user, I think that allowing unknown arguments could be a source of great confusion.

Reply:
Yes, the default should be to crash, but we could support an argument that allows us to "ignore" to, like pydantic does in the setting I linked

Reply:
I'm not against adding an option to parse only known args and return the `Config` object as well as all the remaining arguments. Perhaps a `parse_known` variant to our `parse` function (to allow for the different function signature). But I think that would better fit in another PR.